### PR TITLE
feat: add webpack builder inspector plugin

### PR DESCRIPTION
**Pull request by sanyuan0704**
# PR Details

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/builder/webpack-builder/package.json [hunk 1]
```diff
@@ -66,6 +66,7 @@
   },
   "dependencies": {
     "@modern-js/babel-preset-app": "workspace:*",
+    "@modern-js/inspector-webpack-plugin": "1.0.3",
```
Comment (chenjiahan) [inline packages/builder/webpack-builder/package.json:69]: @modern-js/inspector-webpack-plugin 可以通过 prebundle 优化不

File: packages/builder/webpack-builder/src/plugins/inspector.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+/**
+ * Provide webpack inspector.
+ */
+export const PluginInspector = (): BuilderPlugin => ({
+  name: 'webpack-builder-plugin-inspector',
+
+  setup(api) {
+    api.modifyWebpackChain(async (chain, { CHAIN_ID }) => {
+      const config = api.getBuilderConfig();
+      if (!process.env.WEBPACK_INSPECTOR && !config.tools?.inspector) {
```
Comment (chenjiahan) [inline packages/builder/webpack-builder/src/plugins/inspector.ts:12]: 最好能减少环境变量的使用，因为环境变量不太好管理

File: packages/builder/webpack-builder/package.json (hunk 1 above)
Comment (sanyuan0704) [reply packages/builder/webpack-builder/package.json:69]: 引入这个依赖的时候我加上注释了哈

File: packages/toolkit/utils/src/chainId.ts [hunk 3]
```diff
@@ -108,6 +108,8 @@ export const CHAIN_ID = {
     BANNER: 'banner',
     /** Webpackbar */
     PROGRESS: 'progress',
+    /** Inspector */
+    INSPECTOR: 'inspector',
```
Comment (chenjiahan) [inline packages/toolkit/utils/src/chainId.ts:112]: utils 包需要加个 changeset 哈

File: packages/builder/webpack-builder/src/plugins/inspector.ts (hunk 2 above)
Comment (sanyuan0704) [reply packages/builder/webpack-builder/src/plugins/inspector.ts:12]: 确实，环境比较比较隐蔽，还是只暴露 config 配置吧

File: packages/toolkit/utils/src/chainId.ts (hunk 3 above)
Comment (sanyuan0704) [reply packages/toolkit/utils/src/chainId.ts:112]: 好的

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**